lib/auto_api/states/universal_state.ex

```diff
@@ -0,0 +1,64 @@
+# AutoAPI
+# The MIT License
+#
+# Copyright (c) 2021- High-Mobility GmbH (https://high-mobility.com)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+defmodule AutoApi.UniversalState do
+  @moduledoc """
+  Universal state
+  """
+
+  alias AutoApi.State
+
+  use AutoApi.State, spec_file: "universal.json"
+
+  @type t :: %__MODULE__{
+          vin: State.property(String.t()), # TODO: any spec of VIn?
+          brand: State.property(:atom) # TODO: any spec for brand?
```

Comment:
1. We could add a `@type brand::atom()`, to make it short.
2. We could specify the brands themselves manually:
```
@type brand::
:unknown |
:abarth |
:alfaromeo |
:alpine |
:audi |
:bmw |
:cadillac |
:chevrolet |
:chrysler |
:citroen |
:cupra |
:dacia |
:dodge |
:ds |
:fiat |
:ford |
:honda |
:hyundai |
:iveco |
:jaguar |
:jeep |
:kia |
:lancia |
:land_rover |
:lexus |
:man |
:mazda |
:mercedes_benz |
:mini |
:mitsubishi |
:nissan |
:opel |
:peugeot |
:porsche |
:renault |
:seat |
:skoda |
:smart |
:subaru |
:toyota |
:volkswagen |
:volvo_cars |
:emulator
```

3. We can somehow add it with some metaprogramming, but I don't have the metaprogramming skills to do it right now. 😬 If you have any idea how, `AutoApi.UniversalProperties.brand/0` returns the list of brand atoms as read by the specification file

Reply:
Done with some Macro magics:

```
@type brand :: unquote(Enum.reduce(brands, &{:|, [], [&1, &2]}))
```